src/Rules/Library/ControlShouldSupportTextPatternEditWinform.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Core.Bases;
+using Axe.Windows.Core.Enums;
+using Axe.Windows.Rules.PropertyConditions;
+using Axe.Windows.Rules.Resources;
+using System;
+using static Axe.Windows.Rules.PropertyConditions.ControlType;
+using static Axe.Windows.Rules.PropertyConditions.Framework;
+
+namespace Axe.Windows.Rules.Library
+{
+    [RuleInfo(ID = RuleId.ControlShouldSupportTextPatternEditWinform)]
+    class ControlShouldSupportTextPatternEditWinform : Rule
+    {
+        public ControlShouldSupportTextPatternEditWinform()
+        {
+            this.Info.Description = Descriptions.ControlShouldSupportTextPattern;
+            this.Info.HowToFix = HowToFix.ControlShouldSupportTextPattern;
+            this.Info.Standard = A11yCriteriaId.AvailableActions;
+            this.Info.ErrorCode = EvaluationCode.Error;
+            this.Info.FrameworkIssueLink = "https://aka.ms/FrameworkIssue-ControlShouldSupportTextPatternEditWinForm";
+        }
+
+        public override bool PassesTest(IA11yElement e)
+        {
+            if (e == null) throw new ArgumentNullException(nameof(e));
+
+            return Patterns.Text.Matches(e);
+        }
+
+        protected override Condition CreateCondition()
+        {
+            return Edit & WinForms;
```

Comment:
Can we move this check into a new Condition of `ElementGroups.WinFormsEdit`, then use it from both rules? It's a tiny thing, but if they both point to a shared `Condition`, it's less likely that some future change will accidentally get them out of sync.

Reply:
Sounds good! Done in https://github.com/microsoft/axe-windows/pull/705/commits/d4c4ae2f8d0ac0760a422225725a369f1792014d